libcontainerd/client_local_windows.go

```diff
@@ -418,6 +418,13 @@ func (c *client) createLinux(id string, spec *specs.Spec, runtimeOptions interfa
 		}
 		configuration.NetworkSharedContainerName = spec.Windows.Network.NetworkSharedContainerName
 	}
+	
+	// Add memory limit (if omittet the memory will be limited to default (1GB) in Hyper-V isolation)
+	if spec.Windows.Resources.Memory != nil {
+		if spec.Windows.Resources.Memory.Limit != nil {
```

Comment:
I'm super confused; `spec.Windows.Resources` controls resources for _Linux_ containers? I'm worried that we'll now be duplicating all functionality that's already covered by standard Linux containers; https://github.com/moby/moby/blob/4f5c47aae4105840a16047b7bca424319a7db709/vendor/github.com/opencontainers/runtime-spec/specs-go/config.go#L22-L27

Reply:
I think the distinction Windows / Linux is not so well chosen, possibly HCS / Docker (+plattform specific subset) would be better.
Conversely, the HCS is also a subset of the docker config. 

https://github.com/Microsoft/hcsshim/blob/master/interface.go#L67
https://docs.docker.com/engine/api/v1.35/#operation/ContainerCreate

Reply:
Point is that Linux containers take a whole different set of options for Memory constraints; https://github.com/moby/moby/blob/4f5c47aae4105840a16047b7bca424319a7db709/vendor/github.com/opencontainers/runtime-spec/specs-go/config.go#L273-L289

IIUC, this will now map the Memory-constraints for the _container_ to the Hyper-V VM for the container (not the container itself)

Reply:
Yes that is correct. 
HyperV Isolation mode reserves the memory for the container and the container itself has a limit by container config. 
So in HyperV Isolation the container count is limit by physical memory.